api/main.go

```diff
@@ -1,68 +1,108 @@
 package main
 
 import (
-	"context"
+	"database/sql"
 	"fmt"
 	"log"
-	"math/rand"
 	"net"
-	"time"
+	"os"
+	"strings"
 
+	"github.com/joho/godotenv"
+	_ "github.com/lib/pq"
 	"google.golang.org/grpc"
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
 
+	"github.com/escavelo/pubgolf/api/lib/server"
 	pg "github.com/escavelo/pubgolf/api/proto/pubgolf"
 )
 
 const (
-	PORT = ":50051"
+	DEFAULT_PORT = "50051"
 )
 
-type server struct {
-	pg.UnimplementedAPIServer
-}
+func loadEnv() {
+	env := os.Getenv("PUBGOLF_ENV")
+	if "" == env {
+		env = "dev"
+	}
+
+	log.Printf("Running as PUBGOLF_ENV of %s", env)
 
-func makeImg() string {
-	return fmt.Sprintf("https://via.placeholder.com/640x480/%02x%02x%02x"+
-		"?text=Bottle%%20Open%%20NYC%%202019", rand.Intn(255), rand.Intn(255),
-		rand.Intn(255))
+	// In dev, we assume we're running in the monorepo, in which case the .env is
+	// at the repo root. Otherwise, we're probably in the docker environment and
+	// can access values as real env vars.
+	if strings.ToLower(env) == "dev" {
+		if err := godotenv.Load("../.env"); err != nil {
+			log.Fatal("Error loading .env file")
+		}
+	}
 }
 
-func (s *server) GetSchedule(ctx context.Context,
-	req *pg.GetScheduleRequest) (*pg.GetScheduleReply, error) {
-	log.Printf("Returning schedule for Event Key: %s", req.GetEventKey())
-	if req.GetEventKey() != "nyc-2019" {
-		return nil, status.Error(codes.NotFound, "eventKey was not found")
+func getDbConnectionString() string {
+	user := os.Getenv("DB_USER")
+	password := os.Getenv("DB_PASSWORD")
+	if password != "" {
+		password = ":" + password
+	}
+	host := os.Getenv("DB_HOST")
+	if host == "" {
+		host = "localhost"
```

Comment:
Should these defaults be in a constants or something so they're not hard-coded?

Reply:
Possibly, though localhost:5432 is as close to a default dev setup as possible for Postgres.